static/libs-widget-ext.js

```diff
@@ -0,0 +1,481 @@
+    quickSelectButton.html(lib.name + ' ' + version.version);
+    quickSelectButton.on('click', _.bind(function () {
+        this.selectLibAndVersion(libId, versionId);
+        this.showSelectedLibs();
+        this.onChangeCallback();
+    }, this));
+
+    return libDiv;
+};
+
+LibsWidgetExt.prototype.showFavorites = function () {
+    var favoritesDiv = this.domRoot.find('.lib-favorites');
+    favoritesDiv.html('');
+
+    var faves = this.getFavorites();
+    _.each(faves, _.bind(function (versionArr, libId) {
+        _.each(versionArr, _.bind(function (versionId) {
+            var lib = this.getLibInfoById(libId);
+            if (lib) {
+                var version = lib.versions[versionId];
+                if (version) {
+                    var div = this.newFavoriteLibDiv(libId, versionId, lib, version);
+                    favoritesDiv.append(div);
+                }
+            }
+        }, this));
+    }, this));
+};
+
+LibsWidgetExt.prototype.getAndEmptySearchResults = function () {
+    var searchResults = this.domRoot.find('.lib-results-items');
+    searchResults.html('');
+    return searchResults;
+};
+
+LibsWidgetExt.prototype.newSelectedLibDiv = function (libId, versionId, lib, version) {
+    var template = $('#lib-selected-tpl');
+
+    var libDiv = $(template.children()[0].cloneNode(true));
+
+    var detailsButton = libDiv.find('.lib-name-and-version');
+    detailsButton.html(lib.name + ' ' + version.version);
+    detailsButton.on('click', _.bind(function () {
+        var searchResults = this.getAndEmptySearchResults();
+        this.addSearchResult(libId, lib, searchResults);
+    }, this));
+
+    var deleteButton = libDiv.find('.lib-remove');
+    deleteButton.on('click', _.bind(function () {
+        this.markLibrary(libId, versionId, false);
+        libDiv.remove();
+        this.showSelectedLibs();
+        this.onChangeCallback();
+    }, this));
+
+    return libDiv;
+};
+
+LibsWidgetExt.prototype.conjureUpExamples = function (result, lib) {
+    var examples = result.find('.lib-examples');
+    if (lib.examples.length > 0) {
+        var examplesHeader = $('<b>Examples</b>');
+        var examplesList = $('<ul />');
+        _.each(lib.examples, function (exampleId) {
+            var li = $('<li />');
+            examplesList.append(li);
+            var exampleLink = $('<a>Example</a>');
+            exampleLink.attr('href', '/z/' + exampleId);
+            li.append(exampleLink);
+        });
+
+        examples.append(examplesHeader);
+        examples.append(examplesList);
+    }
+};
+
+LibsWidgetExt.prototype.newSearchResult = function (libId, lib) {
+    var template = $('#lib-search-result-tpl');
+
+    var result = $(template.children()[0].cloneNode(true));
+    result.find('.lib-name').html(lib.name);
+    result.find('.lib-description').html(lib.description ? lib.description : '&nbsp;');
+    result.find('.lib-website-link').attr('href', lib.url ? lib.url : '#');
```

Comment:
ditto a target="_blank"?

Or maybe, this one should be _blank, but the "examples" should replace the current view? I don't want folks to lose their in-progress code, is all...probably both should make a new tab.

Reply:
these external links also need `rel="noopener"`